tests/Feature/AccessTokenControllerTest.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+namespace Laravel\Passport\Tests\Feature;
+
+use Carbon\CarbonImmutable;
+use Illuminate\Contracts\Hashing\Hasher;
+use Illuminate\Database\Eloquent\Factory;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+use Laravel\Passport\Client;
+use Laravel\Passport\ClientRepository;
+use Laravel\Passport\HasApiTokens;
+use Laravel\Passport\Token;
+use Laravel\Passport\TokenRepository;
+use Lcobucci\JWT\Parser;
+
+class AccessTokenControllerTest extends PassportTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        Schema::create('users', function (Blueprint $table) {
+            $table->increments('id');
+            $table->string('email')->unique();
+            $table->string('password');
+            $table->dateTime('created_at');
+            $table->dateTime('updated_at');
+        });
+    }
+
+    protected function tearDown(): void
+    {
+        Schema::dropIfExists('users');
+
+        parent::tearDown();
+    }
+
+    protected function getUserClass(): ?string
+    {
+        return User::class;
+    }
+
+    public function testGettingAccessTokenWithPasswordGrant()
+    {
+        $this->withoutExceptionHandling();
+
+        $password = 'foobar123';
+        $user = new User();
+        $user->email = 'foo@gmail.com';
+        $user->password = $this->app->make(Hasher::class)->make($password);
+        $user->save();
+
+        /** @var Client $client */
+        $client = $this->app->make(Factory::class)->of(Client::class)->state('password_client')->create(['user_id' => $user->id]);
+
+        $response = $this->post(
+            '/oauth/token',
+            [
+                'grant_type' => 'password',
+                'client_id' => $client->id,
+                'client_secret' => $client->secret,
+                'username' => $user->email,
+                'password' => $password,
+            ]
+        );
+
+        $response->assertOk();
+
+        $response->assertHeader('pragma', 'no-cache');
+        $response->assertHeader('cache-control', 'no-store, private');
+        $response->assertHeader('content-type', 'application/json; charset=UTF-8');
+
+        $decodedResponse = $response->decodeResponseJson();
+
+        $this->assertArrayHasKey('token_type', $decodedResponse);
+        $this->assertArrayHasKey('expires_in', $decodedResponse);
+        $this->assertArrayHasKey('access_token', $decodedResponse);
+        $this->assertArrayHasKey('refresh_token', $decodedResponse);
+        $this->assertSame('Bearer', $decodedResponse['token_type']);
+        $expiresInSeconds = 31622400;
+        $this->assertEqualsWithDelta($expiresInSeconds, $decodedResponse['expires_in'], 5);
```

Comment:
@X-Coder264 this line was failing for us randomly. I had to update the value of the `$expiresInSeconds` variable to 31536000 instead. Any idea why? I also don't really know what we're testing here exactly? Why should the `expires_in` be the value of above?

Reply:
@driesvints This tests the expiration time of the issued access token. The expiration time for the password grant is being set here:

https://github.com/laravel/passport/blob/v8.4.0/src/PassportServiceProvider.php#L116

This calls `Passport::tokensExpireIn()` which then returns `new DateInterval('P1Y')` (because there is no `$date` provided and its default value is `null` and `static::$tokensExpireAt` is `null` too).

The difference between our numbers is 86400 seconds which is exactly one day. I guess there is some weird time related stuff going on with that `DateInterval` which causes a random fail of that specific assertion.

Reply:
Did the test maybe start to fail after 29.02.? If so then it is related to the leap year which is exactly why there was a one day difference between our expiration numbers 😄

Reply:
hah, that could be yeah